### ISSUE TITLE
Timestamps are different for gps and accelerometer data.

### ISSUE THREAD

**Issue by davidribs** [umer0586/SensorServer#27]
For context, I'm trying to capture location and accelerometer data at the same time (separate packets) but the timestamp for accelerometer data is nanoseconds since reboot and the time associated with location is unix time. Most importantly, the sampling rate is different so I get about 1 gps coord for every 50 accelerometer reading. I would like for my data to be perfectly synced (I don't need perfect unix time just for both data to be taken at the same time)

TLDR is there some way (that isn't a boge on my end; not lazy btw I just trust a solution you make is more accurate in the end than mine) to get the same timestamp for both gps and accelerometer data (i.e. both unix or both nanoseconds since reboot) in a way that would be precise? 

Appreciate your work btw 👍

>

**Comment by umer0586** [comment]
Unfortunately, it's not possible to have the exact same timestamp for GPS and accelerometer data in the context you described. Android uses different APIs and callbacks for reading location sensors (such as GPS) and motion sensors (such as accelerometer), resulting in different timestamp formats and sampling rates.
